posthog/async_migrations/migrations/0003_fill_person_distinct_id2.py

```diff
@@ -0,0 +1,90 @@
+from functools import cached_property
+
+from ee.clickhouse.client import sync_execute
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperation
+from posthog.constants import AnalyticsDBMS
+from posthog.settings import CLICKHOUSE_DATABASE
+
+"""
+Migration summary:
+
+Schema change to migrate the data from the old person_distinct_id table
+to the new person_distinct_id2 table.
+
+The reason this is needed is for faster `person_distinct_id` queries as the
+old schema worked off of (distinct_id, person_id) pairs, making it expensive
+to for our analytics queries, which need to map from distinct_id -> latest person_id.
+
+The new schema works off of distinct_id columns, leveraging ReplacingMergeTrees
+with a version column we store in postgres.
+
+We migrate teams one-by-one to avoid running out of memory.
+
+The migration strategy:
+
+    1. write to both pdi and pdi2 any new updates (done prior to this migration)
+    2. insert all non-deleted (team_id, distinct_id, person_id) rows from pdi into pdi2 (this migration)
+    3. Once migration has run, we only read/write from/to pdi2.
```

Comment:
In order for reverts to be easier if anything goes wrong perhaps we could write to both pdi & pdi2 for a little while?

Reply:
What is the scenario you are worried about?

For cloud, turning off writes is handled in a separate PR.
For self-hosted, what would be the point? We'd significantly slow ourselves down for it.
If reasoning is failed migrations, see the linked plugin server pr. We only stop writes if this migration is marked a success